plugins/amqp_compressed_proof_plugin/amqp_compressed_proof_plugin.cpp

```diff
@@ -0,0 +1,391 @@
+          )
+         ;
+
+   cli.add_options()
+         ("compressed-proof-delete", bpo::bool_switch(),
+          "Delete all compressed proof data files for unconfirmed messages and blocks");
+}
+
+void amqp_compressed_proof_plugin::plugin_initialize(const variables_map& options) {
+   try {
+      boost::filesystem::path dir = app().data_dir() / "amqp";
+      boost::system::error_code ec;
+      boost::filesystem::create_directories(dir, ec);
+
+      const std::string file_prefix = "cproof";
+
+      chain::controller& controller = app().get_plugin<chain_plugin>().chain();
+
+      if(options.count("compressed-proof-delete") && options.at("compressed-proof-delete").as<bool>())
+         for(const auto& p : boost::filesystem::directory_iterator(dir))
+            if(boost::starts_with(p.path().filename().generic_string(), file_prefix))
+               boost::filesystem::remove(p.path(), ec);
+
+      const boost::filesystem::path reversible_blocks_file = dir / (file_prefix + "-reversible.bin"s);
+
+      std::vector<compressed_proof_generator::result_callback_funcs> callbacks;
+
+      if(options.count("compressed-proof")) {
+         const std::vector<std::string>& descs = options.at("compressed-proof").as<std::vector<std::string>>();
+         std::set<std::string> names_used;
+
+         for(const std::string& desc : descs) {
+            std::vector<std::string> tokens;
+            boost::split(tokens, desc, boost::is_any_of("="));
+            EOS_ASSERT(tokens.size() >= 4, chain::plugin_config_exception, "Did not find 4 tokens in compressed-proof option \"${o}\"", ("o", desc));
+
+            const std::string& name = tokens[0];
+            EOS_ASSERT(name.size(), chain::plugin_config_exception, "Cannot have empty name for compressed-proof \"${o}\"", ("o", desc));
+            EOS_ASSERT(names_used.emplace(name).second,
+                       chain::plugin_config_exception, "Name \"${n}\" used for more than one compressed-proof", ("n", name));
+            std::vector<std::string> filtered_receivers;
+            boost::split(filtered_receivers, tokens[1], boost::is_any_of(","));
+            EOS_ASSERT(filtered_receivers.size(), chain::plugin_config_exception, "Cannot have empty filter list for compressed-proof");
+
+            const boost::filesystem::path amqp_unconfimed_file   = dir / (file_prefix + "-unconfirmed-"s + name + ".bin"s);
+            reliable_amqp_publisher& publisher = my->publishers.emplace_back(tokens[2], tokens[3], "", amqp_unconfimed_file, "eosio.node.compressed_proof_v0");
+
+            std::set<chain::name> filter_on_names;
+            for(const std::string& name_string : filtered_receivers)
+               filter_on_names.emplace(name_string);
+
+            callbacks.emplace_back(std::make_pair(
+               [filter_on_names](const chain::action& act) {
+                  return filter_on_names.find(act.account) != filter_on_names.end();
```

Comment:
I think we do need the ability to filter on the action name as well.  Even if we put the actions we're watching in a separate contract, this contract will likely need some administrative actions that we won't want to include.

Reply:
100% agree; I need this for the bananas and the banana contract. It was something I was going to add later but I'm fine doing it now too